addons/mass_mailing/models/mail_mail.py

```diff
@@ -61,6 +61,7 @@ def send_get_mail_body(self, cr, uid, ids, partner=None, context=None):
         # TDE: temporary addition (mail was parameter) due to semi-new-API
         body = super(MailMail, self).send_get_mail_body(cr, uid, ids, partner=partner, context=context)
         mail = self.browse(cr, uid, ids[0], context=context)
+        body = re.sub('(<!--.*?-->)', '', body, 0, re.DOTALL)
```

Comment:
Well, it solves the issue in the since it gets rid of the comments. This is fine for most comments, excepts the Outlook Conditional CSS (see for example https://templates.mailchimp.com/development/css/outlook-conditional-css/)

By removing the comments, there won't be any distinction anymore. Therefore, I would modify this to at least keep the conditional CSS comments. You should find where are these comments stripped (several replacements are applied to the body).